File: Q10.cpp
#include <iostream>
#include <stdio.h>

using namespace std;

int main()
{
    int idade[20], vezes = 1, pessoas = 0;
    for (int i = 0; i < sizeof idade / sizeof idade[0]; i ++, vezes ++)
    {
        printf("Digite a %dª idade: ", vezes);
        cin >> idade[i];
        if (idade[i] >= 18)
        {
            pessoas ++;
        }
    }
    switch (pessoas)
    {
        case 0:    
            cout << "Das 20 idades que você digitou, nenhuma delas são de pessoas maiores de idade." << endl;
            break;
        case 20:
            cout << "Das 20 idades que você digitou, todas elas são de pessoas maiores de idade." << endl;
            break;
        default:
            printf("Das 20 idades que você digitou, %d delas são de pessoas maiores de idade.\n", pessoas);
            break;
    }
    return 0;
}

File: Q10.java
import java.util.Scanner;

public class Q10
{
    public static Scanner leia = new Scanner(System.in);
    public static void main(String args[])
    {
        int idade[] = new int[20], vezes = 1, pessoas = 0;
        for (int i = 0; i < idade.length; i ++, vezes ++)
        {
            System.out.printf("Digite a %dª idade: ", vezes);
            idade[i] = leia.nextInt();
            if (idade[i] >= 18)
            {
                pessoas ++;
            }
        }
        switch (pessoas)
        {
            case 0:    
                System.out.println("Das 20 idades que você digitou, nenhuma delas são de pessoas maiores de idade.");
                break;
            case 20:
                System.out.println("Das 20 idades que você digitou, todas elas são de pessoas maiores de idade.");
                break;
            default:
                System.out.printf("Das 20 idades que você digitou, %d delas são de pessoas maiores de idade.\n", pessoas);
                break;
        }
    }
}